### PR TITLE
Introduce cache versioning

### PR DESCRIPTION
**Pull request by eduardoj**
Before cache versioning, `cache_key` contained the key and the version.

Introducing cache versioning makes `cache_key` to contain only the key. In order to have the same content as before, we have to use `cache_key_with_version`.

This is a follow up of #9392.

Have a look at this [blog entry](https://blog.bigbinary.com/2019/08/06/rails-adds-support-for-recyclable-cache-keys.html) as reference.

### CODE REVIEW

**Comment by eduardoj** [comment]
[Enable cache versioning (again)](https://trello.com/c/l7nCB7Cf/1453-enable-cache-versioning-again)

File: src/api/config/application.rb [hunk 1]
```diff
@@ -85,8 +85,8 @@ class Application < Rails::Application
 
     config.action_controller.perform_caching = true
 
-    # Don't use cache versioning for now
-    config.active_record.cache_versioning = false
+    # Use cache versioning
```
Comment (dmarcoux) [inline src/api/config/application.rb:88]: [nitpick]

I would remove this comment since it's pretty obvious.

**Comment by dmarcoux** [comment]
Just to be sure, there are 3 instances of `User.session!.cache_key`. Should these be also changed to `User.session!.cache_key_with_version`?

- https://github.com/openSUSE/open-build-service/blob/bbdab04f9c286cdcfec0bad546419f654845e115/src/api/app/models/package.rb#L128
- https://github.com/openSUSE/open-build-service/blob/bbdab04f9c286cdcfec0bad546419f654845e115/src/api/app/views/webui/users/tasks/index.html.haml#L7
- https://github.com/openSUSE/open-build-service/blob/bbdab04f9c286cdcfec0bad546419f654845e115/src/api/app/views/webui/users/tasks/index.html.haml#L22

File: src/api/config/application.rb (hunk 1 above)
Comment (eduardoj) [reply src/api/config/application.rb]: Done.

**Comment by eduardoj** [comment]
> Just to be sure, there are 3 instances of `User.session!.cache_key`. Should these be also changed to `User.session!.cache_key_with_version`?

True! Thanks!

**Comment by hennevogel** [comment]
Why?

**Comment by eduardoj** [comment]
> Why?

Why not?

**Comment by hennevogel** [comment]
> Why not?

Because there are no problems with caching?

**Comment by eduardoj** [comment]
> Because there are no problems with caching?

True, there are currently no problems with caching. Cache versioning is an improved caching concept, introduced in Rails 5.2, enabled by default in Rails 6. Citing the blog post referencing in de PR description "Large applications frequently need to invalidate their cache because cache store has limited memory. We can optimize cache storage and minimize cache miss using recyclable cache keys."

So this pull request doesn't change the current behaviour, but enables cache versioning either for future use cases, or to change current cases.

**Comment by hennevogel** [comment]
> So this pull request doesn't change the current behaviour, but enables cache versioning either for future use cases, or to change current cases.

You should explain this  in ce8daaee58a6f4bfa834cff95ed5e36ca8452fc7

**Comment by eduardoj** [comment]
> > So this pull request doesn't change the current behaviour, but enables cache versioning either for future use cases, or to change current cases.
> 
> You should explain this in [ce8daae](https://github.com/openSUSE/open-build-service/commit/ce8daaee58a6f4bfa834cff95ed5e36ca8452fc7)

Description of the commit changed. Please review again.

**Pull request merged**